plugins/state_history_plugin/include/eosio/state_history_plugin/session.hpp

```diff
@@ -515,7 +537,8 @@ struct session : session_base, std::enable_shared_from_this<session<Plugin, Sock
       current_request = std::move(req);
    }
 
-   void send_update(state_history::get_blocks_request_v0& request, bool fetch_finality_data, state_history::get_blocks_result_v0 result, const chain::signed_block_ptr& block, const chain::block_id_type& id) {
+   template<typename T> // get_blocks_result_v0 or get_blocks_result_v1
+   void send_update(state_history::get_blocks_request_v0& request, bool fetch_finality_data, T result, const chain::signed_block_ptr& block, const chain::block_id_type& id) {
```

Comment:
Should be `T&& result` and caller should `std::move` it.

Reply:
Probably doesn't matter much, but we are using `variant` elsewhere, why switch to `template` and `constexpr` here?

Reply:
The difference in `send_update` for handling `get_blocks_request_v0` and `get_blocks_request_1` is deep inside the middle of code (adding finality_data_log). Using template here seems less messy to try to reuse the same code for both `get_blocks_request_v0` and `get_blocks_request_1`